test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -971,7 +972,17 @@ private ClusterHealthStatus ensureColor(
             // been removed by the master so that the health check applies to the set of nodes we expect to be part of the cluster.
             .waitForNodes(Integer.toString(cluster().size()));
 
-        final ClusterHealthResponse clusterHealthResponse = clusterAdmin().health(healthRequest).actionGet();
+        final ClusterHealthResponse clusterHealthResponse;
+        try {
+            clusterHealthResponse = clusterAdmin().health(healthRequest).get();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            logger.error("interrupted while waiting for health response", e);
+            throw new AssertionError("interrupted while waiting for health response", e);
+        } catch (ExecutionException e) {
+            logger.error("failed to get health response", e);
+            throw new AssertionError("failed to get health response", e);
+        }
```

Comment:
Why did you need this change for your patch? Trying to connect the dots 🤔

Reply:
Sorry should have said - the stack trace in https://github.com/elastic/elasticsearch/issues/105549 is kinda useless because it uses `actionGet()` which strips off the outer `ExecutionException` so you can't see who is treating this exception as a test failure.

Reply:
Ahh, tricky 😌 Thanks for explaining.